### ISSUE TITLE
Memory leak with v6

### ISSUE THREAD

**Issue by denchiklut** [bendotcodes/cookies#419]
After updating to v6, I started having a memory leak issue.
After comparing Heap snapshots, I noticed lots of links in retained objects of `changeListeners in Cookie` 

<img width="1624" alt="Screenshot 2023-11-23 at 11 56 55 PM" src="https://github.com/bendotcodes/cookies/assets/23003955/273917d8-c812-4d58-b28c-f4498b4f74d6">

Downgrading `react-cookie` & `universal-cookie-express` to `5.0.0` solved the memory leak issue.

```sh
 npx envinfo --system --binaries --npmPackages universal-cookie-express,react-cookie,universal-cookie                                ✔ 
  System:
    OS: macOS 13.6
    CPU: (8) arm64 Apple M1 Pro
    Memory: 445.11 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.14.0 - ~/.nvm/versions/node/v18.14.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.14.0/bin/yarn
    npm: 9.3.1 - ~/.nvm/versions/node/v18.14.0/bin/npm
    pnpm: 8.3.1 - ~/.nvm/versions/node/v18.14.0/bin/pnpm
  npmPackages:
    react-cookie: ^6.1.1 => 6.1.1 
    universal-cookie-express: ^6.1.1 => 6.1.1 


```

**Comment by meros** [comment]
I've seen this as well.

**Comment by meros** [comment]
After some investigation I think this is the culprit: https://github.com/bendotcodes/cookies/commit/78165262d1ac4457843e5b2909108a0b1324ff33

The interval keeps an instance to this, that causes the whole cookie class to be retained.

**Comment by Reeywhaar** [comment]
Can confirm. After adding `removeChangeListener` we've fixed our app's memory leak

**Comment by meros** [comment]
My problem is that I am using express middleware, that adds a change listener. It's not possible for me to remove this as I don't have access to the listener function itself.

**Comment by Reeywhaar** [comment]
Then you should address it to the middleware author I guess? Is it available on github?

**Comment by meros** [comment]
https://github.com/bendotcodes/cookies/blob/7fa4ceedbbfa93e0db708b93f7dd712499ffeda6/packages/universal-cookie-express/src/index.ts#L7

**Comment by dawecode** [comment]
Memory leak is happening on our app as well, downgrading has solved it for now.

**Comment by eXon** [comment]
Will be fixed in next release

**Comment by eXon** [comment]
Released in v6.1.2
